src/apps/dashboard/routes/plugins/plugin.tsx

```diff
@@ -0,0 +1,437 @@
+        if (!pluginDetails?.version?.version) return;
+
+        console.debug('[PluginPage] uninstalling plugin', pluginDetails);
+
+        setIsUninstallConfirmOpen(false);
+
+        uninstallPlugin.mutate({
+            pluginId: pluginDetails.id,
+            version: pluginDetails.version.version
+        }, {
+            onSettled: () => {
+                disablePlugin.reset();
+                enablePlugin.reset();
+                installPlugin.reset();
+            }
+        });
+    }, [ disablePlugin, enablePlugin, installPlugin, pluginDetails, uninstallPlugin ]);
+
+    /** Close the uninstall confirmation dialog */
+    const onCloseUninstallConfirmDialog = useCallback(() => {
+        setIsUninstallConfirmOpen(false);
+    }, []);
+
+    useEffect(() => {
+        setIsLoading(isConfigurationPagesLoading || isPackageInfoLoading || isPluginsLoading);
+    }, [ isConfigurationPagesLoading, isPackageInfoLoading, isPluginsLoading ]);
+
+    useEffect(() => {
+        setPluginName(searchParams.get('name') ?? undefined);
+    }, [ searchParams ]);
+
+    useEffect(() => {
+        if (pluginId && !isPluginsLoading) {
+            const pluginInfo = findBestPluginInfo(pluginId, plugins);
+
+            let version;
+            if (pluginInfo) {
+                // Find the installed version
+                const repoVersion = packageInfo?.versions?.find(v => v.version === pluginInfo.Version);
+                version = repoVersion || {
+                    version: pluginInfo.Version,
+                    VersionNumber: pluginInfo.Version
+                };
+            } else {
+                // Use the latest version
+                version = packageInfo?.versions?.[0];
+            }
+
+            let imageUrl;
+            if (pluginInfo?.HasImage) {
+                imageUrl = api?.axiosInstance.getUri({
+                    baseURL: api.basePath,
+                    url: `/Plugins/${pluginInfo.Id}/${pluginInfo.Version}/Image`
+                });
+            }
+
+            setPluginDetails({
+                canUninstall: !!pluginInfo?.CanUninstall,
+                description: pluginInfo?.Description || packageInfo?.description || packageInfo?.overview,
+                id: pluginId,
+                imageUrl: imageUrl || packageInfo?.imageUrl || undefined,
+                isEnabled: (isEnabledOverride && pluginInfo?.Status === PluginStatus.Restart)
+                    ?? pluginInfo?.Status !== PluginStatus.Disabled,
+                name: pluginName || pluginInfo?.Name || packageInfo?.name,
+                owner: packageInfo?.owner,
+                status: pluginInfo?.Status,
+                configurationPage: findBestConfigurationPage(configurationPages || [], pluginId),
+                version,
+                versions: packageInfo?.versions || []
+            });
+        }
+    }, [ api, configurationPages, isEnabledOverride, isPluginsLoading, pluginId, packageInfo, pluginName, plugins ]);
```

Comment:
```suggestion
    const pluginDetails = useMemo(() => {
        if (pluginId && !isPluginsLoading) {
            const pluginInfo = findBestPluginInfo(pluginId, plugins);

            let version;
            if (pluginInfo) {
                // Find the installed version
                const repoVersion = packageInfo?.versions?.find(v => v.version === pluginInfo.Version);
                version = repoVersion || {
                    version: pluginInfo.Version,
                    VersionNumber: pluginInfo.Version
                };
            } else {
                // Use the latest version
                version = packageInfo?.versions?.[0];
            }

            let imageUrl;
            if (pluginInfo?.HasImage) {
                imageUrl = api?.axiosInstance.getUri({
                    baseURL: api.basePath,
                    url: `/Plugins/${pluginInfo.Id}/${pluginInfo.Version}/Image`
                });
            }

            return {
                canUninstall: !!pluginInfo?.CanUninstall,
                description: pluginInfo?.Description || packageInfo?.description || packageInfo?.overview,
                id: pluginId,
                imageUrl: imageUrl || packageInfo?.imageUrl || undefined,
                isEnabled: (isEnabledOverride && pluginInfo?.Status === PluginStatus.Restart)
                    ?? pluginInfo?.Status !== PluginStatus.Disabled,
                name: pluginName || pluginInfo?.Name || packageInfo?.name,
                owner: packageInfo?.owner,
                status: pluginInfo?.Status,
                configurationPage: findBestConfigurationPage(configurationPages || [], pluginId),
                version,
                versions: packageInfo?.versions || []
            };
        }
    }, [api?.axiosInstance, api?.basePath, configurationPages, isEnabledOverride, isPluginsLoading, packageInfo?.description, packageInfo?.imageUrl, packageInfo?.name, packageInfo?.overview, packageInfo?.owner, packageInfo?.versions, pluginId, pluginName, plugins]);
```

Reply:
move the pluginDetails change to line 85
this change reduces unnecessary re-renders.

Before:
![Screenshot 2024-07-14 214731](https://github.com/user-attachments/assets/6eca4220-e4f5-443b-b197-751a7a0e33ac)

After:
![Screenshot 2024-07-14 213816](https://github.com/user-attachments/assets/b6499a0f-11cd-48a9-baa3-7869b0b4649c)